docs/release/handbook.md

```diff
@@ -0,0 +1,288 @@
+
+**Actions for the Release Team:**
+- Request a list of features, from the Working Groups, that require updates to the documentation
+- Ensure the provided components versions match the documentation
+- Work alongside the Working Groups to bring the docs up to date
+
+**Success Criteria:** Documentation for this release completed. A new version
+in the website is cut.
+
+### Distribution testing (3 weeks)
+
+Distribution owners will start testing the latest RC tag and report back issues they find. Depending on the severity of the issues, and number of bug fixes that will accommodate them, the Manifests WG leads might need to cut a new RC tag.
+
+**Actions for other WGs:**
+
+- Evaluate which of the reported issues should be release blocking
+- Work on providing bug fixes for release blocking issues
+- Finalize their provided git revision. It should be stable (not RC) and include fixes for release blocking issues found during this time.
+
+**Actions for Manifests WG:**
+
+- \[OPTIONAL] Create a **vX.Y-rc.4** tag, depending on the number of created issues and bug fixes
+
+**Actions for Release Team:**
+
+- Official Blog post begins drafting and collating information from the Working Groups
+    - (Optional but encouraged) Working Groups start drafting WG-specific blog
+        posts, deep diving into their respective areas
+- Preparation for social media posts can start at the beginning of this phase
+- Release Manager: List the features, and ideally with docs, that made it into the release
+
+**Success Criteria:** Thumbs up from distribution representatives when their testing is complete.
+
+### Release
+
+We made it!
+
+**Actions for Manifests WG:**
+
+- Create the final **vX.Y** tag, pointing to the latest commit of the
+- Create an **vX.Y** tag in <https://github.com/kubeflow/kubeflow>
+- Cut a new GitHub release in <https://github.com/kubeflow/kubeflow>
+
+**Actions for Release Team:**
+
+- Publish release blog post
+- (Optional but encouraged) Working Groups publish individual deep dive blog posts on features or other work they’d like to see highlighted.
+- Publish social media posts
+- Send release announcement to kubeflow-discuss
+
+## Post Release
+
+### Point Release
+Planning for first point release begins. The importance of bugs is left to the
+judgement of the Working Group's tech leads and the Release Manager to decide.
+Fixes included in the point release must satisfy the following criteria:
+* important bug fixes
+* critical security fixes
+* updates to documentation
+
+The tech leads of a WG can also coordinate and decide, alongside the
+Manifests WG tech leads and the Release Manager, if a component's versions should be
+bumped in this point release or go to the next major release.
+
+The Release team continues to meet regularly, as often as needed for the point
+releases. Similar to the major release, the point releases should have a
+tracking issue or board so that distributions can track point releases
```

Comment:
Again, tracking issues and PRs can all be done in a KanBan and an appropriate milestone for the patch release.

Reply:
Indeed, it could be done in a KanBan board and milestones